packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -192,7 +192,12 @@ class IOSDevice extends Device {
   @override
   Future<bool> isAppInstalled(ApplicationPackage app) async {
     try {
-      final RunResult apps = await runCheckedAsync(<String>[_installerPath, '--list-apps']);
+      final RunResult apps = await runCheckedAsync(
+        <String>[_installerPath, '--list-apps'],
+        environment: Map<String, String>.fromEntries(
+          <MapEntry<String, String>>[cache.dyLdLibEntry],
+        ),
+      );
       if (RegExp(app.id, multiLine: true).hasMatch(apps.stdout)) {
```

Comment:
Let's move this outside of the try {} catch {} so that the catch can be narrowed down to catch only `ProcessException`. Let's also add a `printTrace` in the catch. (Then you can also change the return to `return RegExp(...).hasMatch(...)`)

Reply:
Good idea, done.